### PR TITLE
Add GetAppId,TerminateBackgroundApplication desc

### PR DESCRIPTION
**Pull request by gotoextreme**
Signed-off-by: hyunho <hhstark.kang@samsung.com>

### Change Description ###
Add GetAppId,uTerminateBackgroundApplication desc

### CODE REVIEW

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ The main features of the `Tizen.Applications.ApplicationManager` class include:
 
     For installed (but not necessarily running) applications, you can retrieve information with the [Tizen.Applications.ApplicationInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfo.html) class. You can also [retrieve information through a filter](#filter) with the [Tizen.Applications.ApplicationInfoFilter](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfoFilter.html) class.
 
+-   Getting information on current applications
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 2]
```diff
@@ -13,6 +13,11 @@ The main features of the `Tizen.Applications.ApplicationManager` class include:
...
+
+    For current applications, you can retrieve information [manage the current application information](#manage_current) with the [Tizen.Applications](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.Application.html) class.
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 3]
```diff
@@ -40,6 +45,12 @@ To get the application running context and its details, and to operate on the co
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 4]
```diff
@@ -40,6 +45,12 @@ To get the application running context and its details, and to operate on the co
 
 1.  Get the context of the currently-running application by creating an instance of the [Tizen.Applications.ApplicationRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationRunningContext.html) class, with the ID of the application from which the context is being obtained as a parameter.
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 5]
```diff
@@ -40,6 +45,12 @@ To get the application running context and its details, and to operate on the co
...
 
+    If you do not know your App ID, you can retrieve it as follows.
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 6]
```diff
@@ -40,6 +45,12 @@ To get the application running context and its details, and to operate on the co
...
+
+    ```
+    string applicationId = ApplicationManager.GetAppId(Your PID);
+    ```
+
     When an application is not running, it is impossible to get its context.
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 7]
```diff
@@ -113,6 +134,36 @@ To get information on filtered applications:
     }
     ```
 
+<a name="manage_current"></a>
+## Getting Information on Current Applications
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 8]
```diff
@@ -113,6 +134,36 @@ To get information on filtered applications:
...
+
+To get information on current applications:
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 9]
```diff
@@ -113,6 +134,36 @@ To get information on filtered applications:
...
+
+1.  Call the `Current` property of the [Tizen.Applications](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfoFilter.html) class:
+
+    ```
+    Application application = Application.Current;
+    ```
+
+2.  Operate on the application:
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 10]
```diff
@@ -113,6 +134,36 @@ To get information on filtered applications:
...
+
+    -   Get the application version:
+
+        ```
+        string version = application.Version;
+        ```
+
 
 ## Related Information
   * Dependencies
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 11]
```diff
@@ -113,7 +135,37 @@ To get information on filtered applications:
...
+
+To get information on the current application, follow these steps:
+
+1.  Call the `Current` property of the [Tizen.Applications](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfoFilter.html) class:
+
+    ```
+    Application application = Application.Current;
+    ```
+
+2.  Operate on the application context:
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/app-manager.md]: Is this "context" correct in this sentence? In this section it looks like we are talking about Application information itself, not context. Would you please check?

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 12]
```diff
@@ -34,19 +39,26 @@ To enable your application to use the application management functionality:
...
+## Managing Application Running Context
+
+To manage the context of the currently running application, follow these steps:
 
-To get the application running context and its details, and to operate on the context:
+1.  Get the context of the currently running application.
+To get the context, create an instance of the [Tizen.Applications.ApplicationRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationRunningContext.html) class, with the ID of the application from which the context is being obtained as a parameter.
 
-1.  Get the context of the currently-running application by creating an instance of the [Tizen.Applications.ApplicationRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationRunningContext.html) class, with the ID of the application from which the context is being obtained as a parameter.
+    If you do not have an application ID, you can retrieve it as follows.
```
[the remark this answers is not in the document]
Comment (Reni-90) [reply docs/application/dotnet/guides/app-management/app-manager.md]: My intention was to generalize the content. It is absolutely fine to keep it as "the application ID".
@gotoextreme 
Please change "an" to "the". Please change the period (.) to a colon (:). 

If you do not have the application ID, you can retrieve it as:

File: docs/application/dotnet/guides/app-management/app-manager.md (hunk 11 above)
Comment (Reni-90) [reply docs/application/dotnet/guides/app-management/app-manager.md]: I completely agree with you. To be more specific, can we rephrase as:
Operate on the application information:

Please verify.

Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 13]
```diff
@@ -113,7 +135,37 @@ To get information on filtered applications:
     }
     ```
 
+<a name="manage_current"></a>
+## Getting Information on Current Application
+
+To get information on the current application, follow these steps:
+
+1.  Call the `Current` property of the [Tizen.Applications](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfoFilter.html) class:
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

File: docs/application/dotnet/guides/app-management/app-manager.md [hunk 14]
```diff
@@ -13,6 +13,11 @@ The main features of the `Tizen.Applications.ApplicationManager` class include:
 
     For installed (but not necessarily running) applications, you can retrieve information with the [Tizen.Applications.ApplicationInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfo.html) class. You can also [retrieve information through a filter](#filter) with the [Tizen.Applications.ApplicationInfoFilter](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationInfoFilter.html) class.
 
+-   Getting information on the current application
+
+    For current applications, you can retrieve information using the [Tizen.Applications](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.Application.html) class. You can use the retrieved information to [manage the current application](#manage_current).
```
[the remark this answers is not in the document]
Comment (gotoextreme) [reply docs/application/dotnet/guides/app-management/app-manager.md]: Done

**Comment by mijin85cho** [comment]
@gotoextreme Would you please rebase this PR? We have conflicts.


@Reni-90 Would you please re-check below review on line 18?

- **Original :** You can retrieve information manage the current application information
- **reviewed :** you can use the retrieved information to manage the current application information.

In the original sentence, it looks like we only "retrieve" the information.
But in the reviewed sentence, we are going to do something with the retrieved information. (you can **use** ~)
When we read the "Getting Information on Current Application" section, we also see it only gets the information, and do no more further action.
So I'm not sure whether it is correct to link "manage the current application"
 to "Getting information on Current Application".

**Comment by Reni-90** [comment]
@mijin85cho 
Original sentence: You can retrieve information manage the current application information.
The above sentence is ambiguous while reading.

Therefore, I asked @gotoextreme:
"Can we rephrase as:
For current applications, you can retrieve information using the Tizen.Applications class. You can use the retrieved information to manage the current application information."
or
You can retrieve the information to manage the current application information.

I had the same question as per you in my mind. As I was not sure about the rephrase, I asked to verify. When @gotoextreme updated, I thought the rephrase chosen was the correct one. For more information, https://github.com/Samsung/tizen-docs/pull/728#discussion_r289754390

From the original sentence the intention was not clear, hence could not stick to one rephrase. I had to give options. I am sorry if there is confusion.

@gotoextreme 
As per my previous comment, can you verify which one is correct? If none is fine, please let me know the intention so that I can rephrase better.
Thanks.

**Comment by gotoextreme** [comment]
> You can retrieve the information


You can use the retrieved information to manage the current application information."
or
You can retrieve the information to manage the current application information.



I am not able to distinguish a difference between the above two sentences. 
They just have different sequence arent' they?
Both mean that retrieved information will be used to manage the current application.

**Comment by gotoextreme** [comment]
@mijin85cho I resolved the conflict

And about below question.

In the original sentence, it looks like we only "retrieve" the information.
But in the reviewed sentence, we are going to do something with the retrieved information. (you can use ~)
When we read the "Getting Information on Current Application" section, we also see it only gets the information, and do no more further action.
So I'm not sure whether it is correct to link "manage the current application"
to "Getting information on Current Application".

It is about getting information to manage the current application. And it is so obvious that developers need information about the current application to manage the current application that is why I do not describe "more further action". Therefore, the current phrase that (you can use ~) is appropriate for this guide.

**Comment by Reni-90** [comment]
@gotoextreme 
Sorry for the confusion.
Thank you very much for the explanation. This clarifies the doubt.

@mijin85cho 
>It is about getting information to manage the current application. And it is so obvious that developers >need information about the current application to manage the current application that is why I do not >describe "more further action". Therefore, the current phrase that (you can use ~) is appropriate for this >guide.

As per @gotoextreme, the below rephrase is fine.
"For current applications, you can retrieve information using the Tizen.Applications class. You can use the retrieved information to manage the current application information."

Please verify. 
Thank you.

**Comment by mijin85cho** [comment]
@gotoextreme Thanks for the clarification.
If so, I have only one left suggestion about the update.

How about adding a link going to "Getting Information on Current Application" to "retrieve information", not "manage the current application" as below?

For current applications, you can **retrieve information** using the **Tizen.Applications** class. You can use the retrieved information to manage the current application.

**Comment by gotoextreme** [comment]
> Getting Information on Current Application

Done

**Comment by mijin85cho** [comment]
So many thanks, @gotoextreme.
@Reni-90 Would you please review the latest version? I hope you approve this PR after the review if there is no issue, as I'll merge this PR with that approval.

**Pull request merged**